Document plugins

carolynvs opened the pull request:
# What does this change
Initial documentation for the new plugin system

https://deploy-preview-881--porter.netlify.com/mixins/
https://deploy-preview-881--porter.netlify.com/plugins/
https://deploy-preview-881--porter.netlify.com/mixins-vs-plugins/
https://deploy-preview-881--porter.netlify.com/plugins/azure/

# What issue does it fix
The lack of documentation 😎

# Notes for the reviewer
* Is there anything else we want to document while we are feeling spiffy?
* The azure page says to install with `porter plugins install` which hasn't been merged yet. Depending on timing, I can update the page to use the instructions on the readme, which have them build and install manually.

# Checklist
- [ ] Unit Tests
- [x] Documentation
  - [ ] Documentation Not Impacted

vdice commented on docs/content/mixins-vs-plugins.md:
```diff
@@ -0,0 +1,28 @@
+
+[Mixins](/mixins/) are the building blocks for authoring bundles. They
+understand both Porter and CNAB, so you don't have to, and ideally have built in
+extra logic to make your bundles easier to author and more robust.  They
```
nit: consider hyphenating for `built-in`

vdice replied:
nit: consider reducing adjectives in `built in extra logic`

vdice commented on docs/content/mixins-vs-plugins.md:
```diff
@@ -0,0 +1,28 @@
+[Mixins](/mixins/) are the building blocks for authoring bundles. They
+understand both Porter and CNAB, so you don't have to, and ideally have built in
+extra logic to make your bundles easier to author and more robust.  They
+involved when the bundle is built, assisting with generating the Dockerfile for
```
Perhaps `They involved` -> `They are involved`?

vdice commented on docs/content/mixins-vs-plugins.md:
```diff
@@ -0,0 +1,28 @@
+
+[Plugins](/plugins/) **extend** the Porter client itself, **reimplementing**
+Porter's default functionality. There are fixed extension points in Porter with
+a defined interface. For example by default Porter saves claims and credential
```
nit: consider `For example, by default, Porter saves...` or similar

vdice commented on docs/content/mixins-vs-plugins.md:
```diff
@@ -0,0 +1,28 @@
+What both mixins and plugins have in common is that anyone can create their own
+and distribute them, just like the ones that we install with Porter by default.
+The Porter team is committed to making our plugin and mixin ecosystem a level
+playing field with a low barrier to entry. Check our out [Mixin Developer
```
`Check our out` -> `Check out our`

vdice commented on docs/content/mixins/_index.md:
```diff
@@ -1,15 +1,7 @@
-* [aws](/mixins/aws)
-* [gcloud](/mixins/gcloud)
-* [terraform](/mixins/terraform)
+Mixins make Porter special, they are the building blocks that you use when
```
nit: consider adjusting punctuation.  Perhaps `Mixins make Porter special.  They are the...` ?  Or, a semicolon in place of the comma.

vdice commented on docs/content/plugins/azure.md:
```diff
@@ -0,0 +1,72 @@
+
+### Storage
+
+Storage plugins allow Porter to store data, such as claims and credentials, in the Azure's cloud.
```
Perhaps `..in Azure's cloud` or `..in the Azure cloud` instead?

vdice commented on docs/content/mixins-vs-plugins.md:
```diff
@@ -0,0 +1,28 @@
+
+[Plugins](/plugins/) **extend** the Porter client itself, **reimplementing**
+Porter's default functionality. There are fixed extension points in Porter with
+a defined interface. For example, Porter saves claims and credential sets local
```
```suggestion
a defined interface. For example, Porter saves claims and credential sets using the local
```

carolynvs commented:
Gonna hold off on merging this #771 is complete.

dev-drprasad commented:
@carolynvs Looks great to me. Just one suggestion: maybe providing an example for mixins, just like for plugin you did, in mixins-vs-plugins page helps for beginners i think

carolynvs commented:
* Added doc on plugin types: https://deploy-preview-881--porter.netlify.com/plugins/types/
* Added deeper links to the pages and cleaned up existing text about creds to make sure people know about plugins.
